manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -252,6 +252,10 @@ template: |
     - name: {{ $key }}
       value: "{{ $value }}"
     {{- end }}
+    {{- range $key, $val := .Values.global.proxy.env }}
```

Comment:
Why not use ProxyConfig.ProxyMetadata for this purpose ?

Reply:
i.e. ProxyMetadata was introduced so we can deprecate and stop using global - which is a big spaghetti mess of things that are usually not even remotely global.